### ISSUE TITLE
[Meta]: Feature Roadmap

### ISSUE THREAD

**Issue by alexander-alvarez** [offirgolan/ember-light-table#432]
I think we've done a good job of making `ember-light-table` feature rich and extensible. 
Opening this issue to create an open area for discussion around where we see ELT going.

### Large Features Under Pseudo-active Development
- Modular behaviors + Keyboard support https://github.com/offirgolan/ember-light-table/pull/347
- Integrate vertical collection for rendering performance. Work being done [here](https://github.com/alexander-alvarez/ember-light-table/tree/vertical-collection)

### Roadmap
- Freezing columns (after vertical collection is implemented)
- "Out of the box" column and cell types
  - I'd like to see the community pool together their best / most common cell and column patterns and share these with each other so we have an out of the box set, and if the need arises plugin-packs for different columns/cell combos
- ... \<your ideas here\>

Any and all suggestions are welcome.

**Comment by buschtoens** [comment]
> Freezing columns

Can you elaborate a bit, what you mean by that? :)

After integrating `vertical-collection`, proper support for horizontal scrolling (#284) seems like a good next big goal.

Following that, I'd be interested in exploring sticky rows and columns.

**Comment by offirgolan** [comment]
> Modular behaviors + Keyboard support #347
Integrate vertical collection for rendering performance. Work being done here

I think bringing these two should be the MVP for a 2.0 release along with some general refactoring. 

> proper support for horizontal scrolling (#284) seems like a good next big goal.

+1

> Freezing columns (after vertical collection is implemented)

This should be implemented after the first two as it is solely dependent on how we render the rest of the table.

> "Out of the box" column and cell types

These would be good but maybe we can do it in a 3rd party addon? That way others can easily contribute and it will also be forced to use the public API.

**Comment by alexander-alvarez** [comment]
Yup we're all talking & thinking about the same things. I updated the OP

> These would be good but maybe we can do it in a 3rd party addon? That way others can easily contribute and it will also be forced to use the public API.

:heavy_check_mark:  First would be figuring out what all is out there, and then organizing them into addons.
